ParkingLot/Ticket.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+
+namespace ParkingLotSystem
+{
+    public class Ticket
+    {
+        private string carNum;
+        private string ticketNum;
+
+        public Ticket()
+        {
+        }
+
+        public Ticket(string carNum, TicketNumGenerator ticketNumGenerator)
+        {
+            this.CarNum = carNum;
+            this.TicketNum = ticketNumGenerator.GenerateTicketNum();
```

Comment:
ticket number could be generated directly in the constructor without using a GenerateTicketNum class if said class is only used by Ticket class